R/calcIDataElecProd.R

```diff
@@ -0,0 +1,85 @@
+#' calcIDataElecProd
+#'
+#' Use ENERDATA electricity production data to derive OPENPROM input parameter iDataElecProd
+#'
+#' @return  OPENPROM input data iDataElecProd
+#'
+#' @author Anastasis Giannousakis, Fotis Sioutas
+#'
+#' @examples
+#' \dontrun{
+#' a <- calcOutput(type = "IDataElecProd", aggregate = FALSE)
+#' }
+#'
+#' @importFrom dplyr filter %>% mutate select
+#' @importFrom tidyr pivot_wider
+#' @importFrom quitte as.quitte
+
+
+calcIDataElecProd <- function() {
+
+  # load data source (ENERDATA)
+  x <- readSource("ENERDATA", "electricity production", convert = TRUE)
+
+  # filter years
+  fStartHorizon <- readEvalGlobal(system.file(file.path("extdata", "main.gms"), package = "mrprom"))["fStartHorizon"]
+  fStartY <- readEvalGlobal(system.file(file.path("extdata", "main.gms"), package = "mrprom"))["fStartY"]
```

Comment:
I suggest the following code (or similar) to include the latest data, as we did recently on some datasets. 

```
lastYear <- sub("y", "", tail( sort(getYears(x)), 1))
  x <- x[, c(fStartHorizon:lastYear), ]
```